credential_provider/pod_identity_credential_provider.go

```diff
@@ -117,94 +55,55 @@ func (p *podIdentityTokenFetcher) FetchToken(ctx credentials.Context) ([]byte, e
 	}
 
 	// Use the K8s API to fetch the token associated with service account
-	tokRsp, err := p.k8sClient.ServiceAccounts(p.nameSpace).CreateToken(ctx, p.svcAcc, &authv1.TokenRequest{
-		Spec: tokenSpec,
-	}, metav1.CreateOptions{})
+	tokRsp, err := p.k8sClient.ServiceAccounts(p.nameSpace).CreateToken(
+		context.Background(),
+		p.svcAcc,
+		&authv1.TokenRequest{Spec: tokenSpec},
+		metav1.CreateOptions{})
 	if err != nil {
-		return nil, err
+		return "", err
 	}
 
-	return []byte(tokRsp.Status.Token), nil
+	return tokRsp.Status.Token, nil
 }
 
-func (p *PodIdentityCredentialProvider) GetAWSConfig() (*aws.Config, error) {
-	// Get token for Pod Identity
-	token, tokenErr := p.fetcher.FetchToken(context.Background())
-	if tokenErr != nil {
-		return nil, fmt.Errorf("failed to fetch token: %+v", tokenErr)
-	}
-
-	var config *aws.Config
-	var configErr error
-	if p.preferredEndpoint == preferenceIPv4 || p.preferredEndpoint == preferenceAuto {
-		config, configErr = p.getAWSConfigFromPodIdentityAgent(token, podIdentityAgentEndpointIPv4)
-		if configErr != nil {
-			klog.Warningf("IPv4 endpoint attempt failed: %+v.", configErr)
-		} else {
-			return config, nil
-		}
-	}
-
-	if p.preferredEndpoint == preferenceIPv6 || p.preferredEndpoint == preferenceAuto {
-		config, configErr = p.getAWSConfigFromPodIdentityAgent(token, podIdentityAgentEndpointIPv6)
-		if configErr != nil {
-			klog.Warningf("IPv6 endpoint attempt failed: %+v.", configErr)
-		}
-	}
-
-	if configErr != nil {
-		return nil, fmt.Errorf("failed to get AWS config from pod identity agent: %+v", configErr)
-	}
-
-	return config, nil
+// PodIdentityCredentialProvider implements CredentialProvider using pod identity
+type PodIdentityCredentialProvider struct {
+	region             string
+	credentialEndpoint string
+	fetcher            endpointcreds.AuthTokenProvider
+	httpClient         *http.Client
 }
 
-func (p *PodIdentityCredentialProvider) getAWSConfigFromPodIdentityAgent(token []byte, podIdentityAgentEndpoint string) (*aws.Config, error) {
-	req, err := http.NewRequest("GET", podIdentityAgentEndpoint, nil)
```

Comment:
This code is entirely unnecessary in the SDK v2, as the `endpointcreds.Provider{}` handles all of this logic, and can include additional fields like credential expiration time (which is not propagated here, probably a bug)